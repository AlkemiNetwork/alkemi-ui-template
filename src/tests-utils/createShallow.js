import { shallow as _shallow } from "enzyme";

const createShallow = ({ dive }) => {
    const shallow = (node, options) => {
        let wrapper = _shallow(node, options);

        for (let i = 0; i < dive; i++) {
            if (wrapper.name().match(/^WithWidth(.+)$/)) {
                // Diving through a Meterial-UI WidthWith component can not be performed
                // by default because of React.Fragment used in withWidth
                // It is a known issue: https://github.com/airbnb/enzyme/issues/1213
                wrapper = wrapper
                    .dive()
                    .children()
                    .at(0);
            } else {
                wrapper = wrapper.dive();
            }
        }
        return wrapper;
    };

    return shallow;
};

export default createShallow;
