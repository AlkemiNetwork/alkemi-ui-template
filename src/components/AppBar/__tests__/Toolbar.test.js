import React from "react";
import toJson from "enzyme-to-json";

import { MuiThemeProvider, createMuiTheme } from "@material-ui/core/styles";

import AppBarToolbar from "../Toolbar";
import createShallow from "../../../tests-utils/createShallow";

describe("<AppBarToolbar />", () => {
    let shallow;
    let wrapper;

    beforeAll(() => {
        shallow = createShallow({ dive: 4 });
    });

    describe("when shifted=true and width is smDown", () => {
        beforeEach(() => {
            global.window.innerWidth = 959;
            wrapper = shallow(
                <MuiThemeProvider
                    theme={createMuiTheme({
                        breakpoints: { values: { md: 960 } }
                    })}
                >
                    <AppBarToolbar shifted />
                </MuiThemeProvider>
            );
        });

        it("matches snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it("has props disableGutters=false", () => {
            expect(wrapper.props().disableGutters).toBeFalsy();
        });
    });

    describe("when shifted=true andd width is mdUp", () => {
        beforeEach(() => {
            global.window.innerWidth = 960;
            wrapper = shallow(
                <MuiThemeProvider
                    theme={createMuiTheme({
                        breakpoints: { values: { md: 960 } }
                    })}
                >
                    <AppBarToolbar shifted />
                </MuiThemeProvider>
            );
        });

        it("matches snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it("has props disableGutters=true", () => {
            expect(wrapper.props().disableGutters).toBe(true);
        });
    });

    describe("when shifted=false and width is smDown", () => {
        beforeEach(() => {
            global.window.innerWidth = 959;
            wrapper = shallow(
                <MuiThemeProvider
                    theme={createMuiTheme({
                        breakpoints: { values: { md: 960 } }
                    })}
                >
                    <AppBarToolbar />
                </MuiThemeProvider>
            );
        });

        it("matches snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it("has props disableGutters=true", () => {
            expect(wrapper.props().disableGutters).toBe(true);
        });
    });
});
