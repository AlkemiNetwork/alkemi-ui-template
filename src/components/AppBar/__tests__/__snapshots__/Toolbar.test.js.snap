// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBarToolbar /> when shifted=false and width is smDown matches snapshot 1`] = `
<Toolbar
  classes={
    Object {
      "dense": "MuiToolbar-dense-4",
      "gutters": "MuiToolbar-gutters-2",
      "regular": "MuiToolbar-regular-3",
      "root": "MuiToolbar-root-1",
    }
  }
  disableGutters={true}
  variant="regular"
>
  <Connect(WithStyles(AppbarButton)) />
  <Connect(AppBarTitle) />
</Toolbar>
`;

exports[`<AppBarToolbar /> when shifted=true and width is smDown matches snapshot 1`] = `
<Toolbar
  classes={
    Object {
      "dense": "MuiToolbar-dense-4",
      "gutters": "MuiToolbar-gutters-2",
      "regular": "MuiToolbar-regular-3",
      "root": "MuiToolbar-root-1",
    }
  }
  disableGutters={false}
  variant="regular"
>
  <Connect(WithStyles(AppbarButton)) />
  <Connect(AppBarTitle) />
</Toolbar>
`;

exports[`<AppBarToolbar /> when shifted=true andd width is mdUp matches snapshot 1`] = `
<Toolbar
  classes={
    Object {
      "dense": "MuiToolbar-dense-4",
      "gutters": "MuiToolbar-gutters-2",
      "regular": "MuiToolbar-regular-3",
      "root": "MuiToolbar-root-1",
    }
  }
  disableGutters={true}
  variant="regular"
>
  <Connect(WithStyles(AppbarButton)) />
  <Connect(AppBarTitle) />
</Toolbar>
`;
